File: Hot 100/49. 字母异位词分组.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string> > ans;
        unordered_map<string, vector<string> > mp;
        for(auto& str: strs) {
            string count(26, '0');
            for(auto& c: str){
                count[c-'a']+=1;
            }
            mp[count].push_back(str);
        }
        for(auto& [k, v]: mp){
            ans.push_back(v);
        }
        return ans;
    }
};

File: 刷题/golang/49. 字母异位词分组.go
func groupAnagrams(strs []string) [][]string {
    mp:=map[[26]int][]string{}
    for _,str:= range strs {
        count:=[26]int{}
        for _,ch:= range str {
            count[ch-'a']++
        }
        mp[count]=append(mp[count],str)
    }
    ans:=make([][]string,0)
    for _,v:=range mp {
        ans=append(ans,v)
    }
    return ans
}